src/article-controller.js

```diff
@@ -0,0 +1,72 @@
+(function(exports){
+
+  function ArticleController(articleListView = new ArticleListView , singleArticleView = new SingleArticleView){
+    this._articleListView = articleListView;
+    this._singleArticleView = singleArticleView;
+  }
+
+
+  ArticleController.prototype = {
+    getElementOf: function(id){ return document.getElementById(id).innerHTML;},
+
+    changeContent: function(content){
+      document
+      .getElementById('app')
+      .innerHTML = content;},
+
+    guardianRequest: function(link){
+      var self = this
+     var request = new XMLHttpRequest();
+     request.open('GET', link);
+     request.onload = function (){
+       var data = JSON.parse(request.responseText).response.results;
+       data.forEach(function(article){ self.aylienRequest(article, aylien.request(article.webUrl),  function(){self.loadData(); self.hashChange()})});
+       }
+     request.send();
+   },
+
+   aylienRequest: function(data, link,callBack){
+     var self = this;
+     var request = new XMLHttpRequest();
+     request.open('GET', link);
+     request.onload = function (){
+       var body = JSON.parse(request.responseText);
+        self._articleListView.addArticle(data, body)
+        callBack();
+       }
+     request.send();
+   },
+
+    guardian: function(){
+      this.guardianRequest(guardian.request)},
+
+    loadData: function(){
+      var self = this;
+      document
+      .getElementById('app')
+      .innerHTML = self._articleListView.toHtml();
+      self.hashChange();
+    },
+
+    hashChange: function() {
+      var self = this;
+      var content = this._articleListView._list;
+      function showArticleForCurrentPage() {
```

Comment:
The hash change event looks like any hash change assumes that you are navigating to the single article view. What if somebody navigates back to the main article list?

A cleaner implementation uses a route table approach, e.g.

```
var routes = [{
  route: /articles/, method: showArticleList,
  route: /articles\/(\d+)/, method: showArticle,
}]

var method = routes.find((route) => {
  return route.test(window.location);
});

method();
```

This method uses regular expressions to match the url (see https://eloquentjavascript.net/09_regexp.html if you are not familiar with these, I realise we haven't covered them on the course). You don't have to use regular expressions, but the point is that you should have a list of routes and a list of controller methods that match them, and then execute the method based on which route matches.

NB I didn't think very long about the code above so it may have a bug or two...